tests/Integration/PrestaShopBundle/EventListener/AccessDeniedListenerTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\PrestaShopBundle\EventListener;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\Integration\Utility\LoginTrait;
+
+class AccessDeniedListenerTest extends WebTestCase
+{
+    use LoginTrait;
+
+    protected KernelBrowser $client;
+    protected Router $router;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->client = self::createClient();
+        $this->router = self::$kernel->getContainer()->get('router');
+    }
+
+    public function testAccessDenied(): void
+    {
+        $this->loginUser($this->client);
+
+        $this->client->request('GET', $this->router->generate('test_index'));
+
+        $response = $this->client->getResponse();
+        $this->assertEquals(Response::HTTP_FORBIDDEN, $response->getStatusCode());
+    }
+
+    public function testValidAccess(): void
+    {
+        $this->loginUser($this->client);
+
+        $this->client->request('GET', $this->router->generate('test_something_complex'));
+
+        $response = $this->client->getResponse();
+        $this->assertEquals(Response::HTTP_OK, $response->getStatusCode());
+    }
+
+    public function testRedirection(): void
+    {
+        $this->loginUser($this->client);
+
+        $this->client->request('GET', $this->router->generate('test_redirect'));
+
+        $response = $this->client->getResponse();
+        $this->assertEquals(Response::HTTP_OK, $response->getStatusCode());
```

Comment:
You should have a redirection in this case no? I believe by default the test client follows redirections (at least it is configured to follow them by default when no error occurs if I remember well).

To correctly test this redirection feature you should probably disable the automatic redirection following, and check that the response has the correct HTTP status code along with the proper location header

Note: this could be performed in two different tests, as this test following automatic redirection is also relevant, but having another one that checks the intermediate redirection is also important

Reply:
problem solved. 2 routes had the same name, it stops on the 1st, which had the right role and a 200 status code